programs/stake/src/stake_state.rs

```diff
@@ -1000,10 +1022,16 @@ struct ValidatedDelegatedInfo {
 fn validate_delegated_amount(
     account: &BorrowedAccount,
     meta: &Meta,
+    feature_set: &FeatureSet,
 ) -> Result<ValidatedDelegatedInfo, InstructionError> {
     let stake_amount = account
         .get_lamports()
         .saturating_sub(meta.rent_exempt_reserve); // can't stake the rent
+    if feature_set.is_active(&stake_allow_zero_undelegated_amount::id())
```

Comment:
Perhaps add a comment about the behavior change here?
Maybe something like this (but moar better): Previously, all StakeState::Initialize accounts were required to have a balance greater than the minimum delegation amount. After feature `stake_allow_zero_undelegated_amount`, StakeState::Initialize accounts may have a lower balance, so the minimum must be checked on via this function on delegation.

Reply:
Is this feature gate actually necessary? It's not currently possible to initialize a stake account which doesn't already hit the minimum delegation.

Reply:
Never mind, I see it was possible to withdraw and then delegate again. Will make sure this behavior gets documented here: https://github.com/solana-labs/solana/pull/24603